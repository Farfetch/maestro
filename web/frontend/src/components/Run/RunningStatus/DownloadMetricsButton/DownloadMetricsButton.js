import { Button } from "antd";

import { runMetricsDownloadUrl } from "../../../../lib/routes";

const DownloadMetricsButton = ({ runId }) => (
  <a
    href={runMetricsDownloadUrl(runId)}
    target="_blank"
    download
    rel="noreferrer"
  >
    <Button type="primary">Download metrics</Button>
  </a>
);

export default DownloadMetricsButton;
