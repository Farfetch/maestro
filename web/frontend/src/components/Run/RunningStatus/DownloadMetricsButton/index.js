import DownloadMetricsButton from "./DownloadMetricsButton";

export default DownloadMetricsButton;
