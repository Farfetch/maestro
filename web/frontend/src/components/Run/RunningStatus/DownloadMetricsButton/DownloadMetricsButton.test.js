import { render } from "@testing-library/react";
import React from "react";

import DownloadMetricsButton from "./DownloadMetricsButton";

describe("components/Run/RunningStatus/DownloadMetricsButton", () => {
  test(`should render DownloadMetricsButton component`, async () => {
    const rendered = render(<DownloadMetricsButton runId="1-2-3" />);

    const badgeComponent = rendered.container;
    expect(badgeComponent.outerHTML).toMatchSnapshot();
  });
});
