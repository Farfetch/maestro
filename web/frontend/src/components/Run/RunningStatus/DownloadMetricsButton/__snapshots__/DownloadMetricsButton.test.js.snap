// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Run/RunningStatus/DownloadMetricsButton should render DownloadMetricsButton component 1`] = `"<div><a href=\\"http://api.maestro.io/api/run_metrics/1-2-3/download\\" target=\\"_blank\\" download=\\"\\" rel=\\"noreferrer\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Download metrics</span></button></a></div>"`;
